apps/bfd-server/bfd-server-war/src/test/java/gov/cms/bfd/server/war/AccessJsonE2E.java

```diff
@@ -0,0 +1,123 @@
+package gov.cms.bfd.server.war;
+
+import static io.restassured.RestAssured.given;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import gov.cms.bfd.model.rif.entities.Beneficiary;
+import gov.cms.bfd.server.sharedutils.BfdMDC;
+import gov.cms.bfd.server.war.commons.TransformerConstants;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Verifies that access.json is written to as expected. */
+public class AccessJsonE2E extends ServerRequiredTest {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(AccessJsonE2E.class);
+
+  /** Verifies that access.json is written to within BFD-server-war via API call. */
+  @Test
+  public void verifyAccessJson() throws IOException {
+
+    Beneficiary beneficiary = testUtils.getFirstBeneficiary(testUtils.loadSampleAData());
+    String mbiHash = beneficiary.getMbiHash().orElseThrow();
+    String requestString =
+        baseServerUrl
+            + "/v2/fhir/Patient/?identifier="
+            + TransformerConstants.CODING_BBAPI_BENE_MBI_HASH
+            + "|"
+            + mbiHash;
+
+    Path accessLogJson =
+        ServerTestUtils.getWarProjectDirectory()
+            .resolve("target")
+            .resolve("server-work")
+            .resolve("access.json");
+
+    // Empty the access json to avoid pollution from other tests
+    try (BufferedWriter writer = Files.newBufferedWriter(accessLogJson)) {
+      writer.write("");
+      writer.flush();
+    }
+
+    given()
+        .spec(requestAuth)
+        .contentType("application/json")
+        .expect()
+        .statusCode(200)
+        .when()
+        .get(requestString);
+
+    // Verify that the access JSON is working, as expected.
+    try {
+      TimeUnit.MILLISECONDS.sleep(1000); // Needed to resolve a race condition
+    } catch (InterruptedException ignored) {
+    }
```

Comment:
This is a curious few lines. I see _this_ implementation requires that we not only maintain the solution, but we double the value. It wasn't _too_ offensive when we originally accepted this as part of #1557, but I do recall pairing on this and asking if it could be removed... I **think** it was only kept because we couldn't get GitHub Actions to behave without it at the time. Is that where you saw failures or were you seeing these failures locally, too?

In general, can we say that we're more tolerant of sleeps like this in tests?

Reply:
I didn't see any issues removing this when I was testing, but it existed before and I didn't want the test failing in a week cause I didn't locally see the issue a few times. If you're thinking the original cause is fixed, I can nix it, since I don't like thread sleeps either.

Reply:
Removed it, as long as github actions passes for this PR, I'll assume it'll be fine. (If not, can add it back if we see an issue)